pal/pal.h

```diff
@@ -0,0 +1,107 @@
+
+#ifndef PAL_H
+#define PAL_H
+
+typedef enum {
+    PAL_OK,
+    PAL_ERROR_OUT_OF_MEMORY,
+    PAL_ERROR_ARG,
+    PAL_ERROR_UNABAILABLE
+} pal_result;
+
+typedef enum {
+    PAL_LED_OFF,
+    PAL_LED_RED,
+    PAL_LED_YELLOW,
+    PAL_LED_GREEN
+} pal_led_color;
+
+/**
+ * @brief  Get a string with the current pal version.
+ *
+ * @return The current PAL version.
+ */
+const char* pal_version_get_string(void);
+
+/**
+ * @brief  Get an integer with the major of the current pal version.
+ *
+ * @return The major of the current PAL version.
+ */
+const int pal_version_get_major(void);
+
+/**
+ * @brief  Get an integer with the minor of the current pal version.
+ *
+ * @return The minor of the current PAL version.
+ */
+const int pal_version_get_minor(void);
+
+/**
+ * @brief  Get an integer with the patch of the current pal version.
+ *
+ * @return The patch of the current PAL version.
+ */
+const int pal_version_get_patch(void);
+
+/**
+ * @brief  Get the current temperature on the MCU
+ *
+ * @param[out] temperature      The pointer to 'unsigned short' to return the current temperature in a scale from 0 to 65535.
+ *
+ * @return The 'pal_result' with the operation result.
```

Comment:
nit: you can use `#` to resolve the type in doxygen. 
```suggestion
 * @return The #pal_result with the operation result.
```